house-robber/minji-go.java

```diff
@@ -0,0 +1,11 @@
+class Solution {
+    public int rob(int[] nums) {
+        int[] sum = new int[nums.length+1];
+        sum[0] = nums[0];
+        if(nums.length>1) sum[1] = Math.max(nums[0], nums[1]);
+        for(int i=2; i<nums.length; i++){
+            sum[i]=Math.max(nums[i]+sum[i-2],sum[i-1]);
```

Comment:
안녕하세요~
점화식을 잘 살펴보면 `sum`배열 중에서 `sum[i], sum[i-1], sum[i-2]` 세 요소만 사용한다는 사실을 알 수 있는데요
아직 시간이 좀 남았으니 위 사실을 이용해서 공간복잡도 최적화를 도전해보시면 어떨까요? :)

Reply:
@obzva 안녕하세요~ 
공간복잡도 최적화할 생각을 못했는데 알려주셔서 감사합니다 :) 

1. 배열의 요소를 3개로 하고, %연산 사용
```suggestion
            sum[i%3]=Math.max(nums[i]+sum[(i-2)%3],sum[(i-1)%3]);
```
2. 배열대신 변수와 swap 사용
```suggestion
            int sum3=Math.max(nums[i]+sum1,sum2);
            sum1=sum2;
            sum2=sum3;
```

이렇게 두가지 방식을 생각해봤는데, 어떤 게 더 좋은 코드라고 볼 수 있을까요?

Reply:
저는 두번째가 더 직관적인 것 같아요 ㅎㅎㅎ!